server/src/main/java/org/opensearch/telemetry/tracing/TracerFactory.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.telemetry.Telemetry;
+import org.opensearch.telemetry.TelemetrySettings;
+import org.opensearch.telemetry.tracing.noop.NoopTracer;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Optional;
+
+/**
+ * TracerManager represents a single global class that is used to access tracers.
+ *
+ * The Tracer singleton object can be retrieved using tracerManager.getTracer(). The TracerManager object
+ * is created during class initialization and cannot subsequently be changed.
+ */
+public class TracerFactory implements Closeable {
+
+    private static final Logger logger = LogManager.getLogger(TracerFactory.class);
+
+    private volatile Tracer defaultTracer;
+    private final Object mutex = new Object();
+    private final TelemetrySettings telemetrySettings;
+    private final Optional<Telemetry> telemetry;
+    private final ThreadContext threadContext;
+
+    public TracerFactory(TelemetrySettings telemetrySettings, Optional<Telemetry> telemetry, ThreadContext threadContext) {
+        this.telemetrySettings = telemetrySettings;
+        this.telemetry = telemetry;
+        this.threadContext = threadContext;
+    }
+
+    /**
+     * Returns the tracer instance
+     * @return tracer instance
+     */
+    public Tracer getTracer() {
+        return isTracingDisabled() || !telemetry.isPresent() ? NoopTracer.INSTANCE : getOrCreateDefaultTracerInstance();
```

Comment:
We need to clean up `TracerFactory` a bit, checking `isTracingDisabled()` on each `getTracer()` invocation is probably fine (since this is a dynamic cluster setting) but not `telemetry` - it is either present or not, we don't need to store optional:
```
private final TracingTelemetry tracingTelemetry;
```

```
        this.tracingTelemetry = telemetry.map(Telemetry::getTracingTelemetry).orElse(NoopTracer.INSTANCE);

```

Reply:
Done, this makes TracerFactory further clean!